app/subjects/index.cjsx

```diff
@@ -28,28 +30,38 @@ module?.exports = React.createClass
   getInitialState: ->
     subject: null
     tab: 0
+    comments: []
+    commentsMeta: {}
+
+  getDefaultProps: ->
+    query: page: 1
 
   componentWillMount: ->
-    @setSubject()
+    @setSubject().then(@setComments)
 
   componentWillReceiveProps: (nextProps) ->
     if nextProps.params?.id isnt @props.params?.id
-      @setSubject()
+      @setSubject().then(@setComments)
+
+    if nextProps.query.page isnt @props.query.page
+      @setComments(@state.subject, nextProps.query.page)
 
   setSubject: ->
     subjectId = @props.params?.id.toString()
     apiClient.type('subjects').get(subjectId)
       .then (subject) =>
         @setState {subject}
+        subject
+
+  setComments: (subject = @state.subject, page = @props.query.page ? 1) ->
+    talkClient.type('comments')
+      .get({focus_id: subject.id, focus_type: 'Subject', page})
+      .then (comments) =>
+        commentsMeta = comments[0]?.getMeta()
+        @setState {comments, commentsMeta}
 
-  comment: (data, i) ->
-    <CommentLink key={data.id} comment={data}>
-      <div className="talk-module">
-        <strong>{data.user_display_name}</strong>
-        <br />
-        <Markdown project={@props.project}>{data.body}</Markdown>
-      </div>
-    </CommentLink>
+  comment: (comment, i) ->
+    <Comment data={comment} locked={true} linked={true} />
```

Comment:
Just added the :key: prop.
`locked` hides the reply/edit/report buttons, and `linked` adds in that extra link,  which I think always make sense on the subject page in it's current state 

That could change with #1271